sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -436,3 +436,21 @@ case class Murmur3Hash(children: Seq[Expression], seed: Int) extends Expression
     }
   }
 }
+
+/**
+  * An expression that will print the value of child to stderr (used for debugging codegen).
+  */
+case class Echo(child: Expression) extends UnaryExpression {
```

Comment:
So this is only used if you construct the expressions manually?

I'd rename this to PrintToStderr. Echo means it is returning whatever you are typing in, which is not what this is about.